setup.py

```diff
@@ -636,6 +636,12 @@ def build_mx_extension(build_ext, options):
         mxnet_mpi_lib.define_macros += [('MSHADOW_USE_CUDA', '1')]
     else:
         mxnet_mpi_lib.define_macros += [('MSHADOW_USE_CUDA', '0')]
+    
+    if os.environ.get('MXNET_USE_MKLDNN'):
```

Comment:
Can we automatically detect if mkldnn is enabled in MXNet? Using an env variable is not very good user experience. We added [runtime feature](https://github.com/apache/incubator-mxnet/commit/3a6fe22b660797f0b6f7d0530c604024b6f84bae) discovery API recently which might be able to used here.